File: docs/breez-sdk/snippets/wasm/messages.ts
import { type BreezClient, verifyMessage } from '@breeztech/breez-sdk-spark'

const exampleSignMessage = async (client: BreezClient) => {
  // ANCHOR: sign-message
  // Set to true to get a compact signature rather than a DER
  const compact = true

  const signMessageResponse = await client.message.sign({
    message: '<message to sign>',
    compact
  })

  const signature = signMessageResponse.signature
  const pubkey = signMessageResponse.pubkey

  console.log(`Pubkey: ${pubkey}`)
  console.log(`Signature: ${signature}`)
  // ANCHOR_END: sign-message
}

const exampleCheckMessage = async () => {
  // ANCHOR: check-message
  const checkMessageResponse = verifyMessage({
    message: '<message>',
    pubkey: '<pubkey of signer>',
    signature: '<message signature>'
  })
  const isValid = checkMessageResponse.isValid

  console.log(`Signature valid: ${isValid}`)
  // ANCHOR_END: check-message
}

File: docs/breez-sdk/snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/Messages.kt
package com.example.kotlinmpplib

import breez_sdk_spark.*

class Messages {
    suspend fun signMessage(sdk: BreezSdk) {
        // ANCHOR: sign-message
        val message = "<message to sign>"
        // Set to true to get a compact signature rather than a DER
        val compact = true
        try {
            val signMessageRequest = SignMessageRequest(message, compact)
            val signMessageResponse = sdk.signMessage(signMessageRequest)

            val signature = signMessageResponse?.signature
            val pubkey = signMessageResponse?.pubkey

            // Log.v("Breez", "Pubkey: ${pubkey}")
            // Log.v("Breez", "Signature: ${signature}")
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: sign-message
    }

    suspend fun checkMessage(sdk: BreezSdk) {
        // ANCHOR: check-message
        val message = "<message>"
        val pubkey = "<pubkey of signer>"
        val signature = "<message signature>"
        try {
            val checkMessageRequest = CheckMessageRequest(message, pubkey, signature)
            val checkMessageResponse = sdk.checkMessage(checkMessageRequest)

            val isValid = checkMessageResponse?.isValid

            // Log.v("Breez", "Signature valid: ${isValid}")
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: check-message
    }

}
